sdk/java-sdk/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java

```diff
@@ -280,13 +290,19 @@ public interface Step {
     final private String _name;
     final public Function<CallInput, DeferredCall<DefCallInput, DefCallOutput>> callFunc;
     final public Function<DefCallOutput, Effect.TransitionalEffect<Void>> transitionFunc;
+    final public Class<CallInput> callInputClass;
+    final public Class<DefCallOutput> transitionInputClass;
 
     public CallStep(String name,
                     Function<CallInput, DeferredCall<DefCallInput, DefCallOutput>> callFunc,
                     Function<DefCallOutput, Effect.TransitionalEffect<Void>> transitionFunc) {
       _name = name;
       this.callFunc = callFunc;
       this.transitionFunc = transitionFunc;
+      Class<?>[] callClasses = TypeResolver.resolveRawArguments(Function.class, callFunc.getClass());
+      Class<?>[] transitionClasses = TypeResolver.resolveRawArguments(Function.class, transitionFunc.getClass());
+      this.callInputClass = (Class<CallInput>) callClasses[0];
```

Comment:
Again. I think this may be my ignorance of implementation details. How do you know `callClasses` are not empty and if there are multiple results you only need the first?

Reply:
It is quite simple, you have a `Function` which is parametrized with 2 types, so you know it won't be empty, and you need the first one (they are ordered).

Reply:
That makes sense 👍